models.py

```diff
@@ -194,7 +194,7 @@ def __init__(self, controller, **kwargs):
         super().__init__()
 
         # required attributes
-        self._id = kwargs.get('id', uuid4().hex)
+        self._id = kwargs.get('id', uuid4().hex[16:])
```

Comment:
Is there any specific reason to get the _last_ 16 random digits? If not, I think getting the first 16 chars is somehow more straightforward than getting the last ones. Like: "get the first ones, ignore the rest" instead of "jump through these 16 ones that'll be used later for something".

```suggestion
        self._id = kwargs.get('id', uuid4().hex[:16])
```